### PR TITLE
SALTO-2633: add dependency changer between field that modified to a new context - Jira

### PR DESCRIPTION
**Pull request by shayc331**
_Add dependency changer between field that modified to a new context_

---

_When a change validator omitted an adding of a context, the dependency caused omitting the modification change as well_

---
_Release Notes_: 
_Jira_
* fixed a bug related to the dependency between adding a new field context to modifying the field

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/test/dependency_changers/field_contexts.test.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { ObjectType, InstanceElement, ElemID, toChange, ReferenceExpression, CORE_ANNOTATIONS } from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import { fieldContextDependencyChanger } from '../../src/dependency_changers/field_contexts'
+import { JIRA } from '../../src/constants'
+import { FIELD_CONTEXT_TYPE_NAME, FIELD_TYPE_NAME } from '../../src/filters/fields/constants'
+
+describe('fieldContextsDependencyChanger', () => {
```
Comment (alonstern) [inline packages/jira-adapter/test/dependency_changers/field_contexts.test.ts:22]: There are two branches not covered in the tests:
- When dependencies.get return undefined in https://github.com/salto-io/salto/pull/3350/files#diff-ab14a39d534fce81df69c28f356d2a723571d6ca639df11e661fe52c8d01b3a8R30
- When `change` is undefined https://github.com/salto-io/salto/pull/3350/files#diff-ab14a39d534fce81df69c28f356d2a723571d6ca639df11e661fe52c8d01b3a8R33

Please add tests for them

**Review by alonstern — COMMENTED** [review]
🔥 🔥

File: packages/jira-adapter/test/dependency_changers/field_contexts.test.ts (hunk 1 above)
Comment (shayc331) [reply packages/jira-adapter/test/dependency_changers/field_contexts.test.ts:22]: Done

**Review by alonstern — APPROVED** [review]
👌👌

**Pull request merged**